lib/components/molecules/Accordion/Accordion.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+/**
+ *
+ * Accordion
+ *
+ */
+import React, { useState } from 'react';
+import type { Node } from 'react';
+import styled from 'styled-components';
+import styles from './Accordion.style';
+import type { AccordionHeaderProps, AccordionPanelProps, AccordionProps } from './types';
+
+const AccordionHeader = ({
+  children,
+  className,
+  ariaLabelAttr,
+  accordionPanelId,
+}: AccordionHeaderProps): Node => {
+  const [ariaExpandedVal, toggleAriaExpandedOnClick] = useState(false);
+
+  const handleClick = () => toggleAriaExpandedOnClick(!ariaExpandedVal);
+
+  return (
+    <summary
+      className={className}
+      role="button"
+      aria-expanded={ariaExpandedVal}
+      aria-label={ariaLabelAttr}
+      aria-labelledby={accordionPanelId}
+      onClick={handleClick}
+      onKeyDown={handleClick}
+      tabIndex="0"
+    >
+      {children}
+    </summary>
+  );
+};
+
+AccordionHeader.defaultProps = {
+  className: 'accordion-header',
+  ariaLabelAttr: 'aria-label',
+};
+
+/**
+ * AccordionPanel
+ *
+ */
+const AccordionPanel = ({ children, className, id }: AccordionPanelProps): Node => {
+  const childrenWithProps = React.Children.map(children, accordionheader => {
+    if (accordionheader.type.displayName === AccordionHeader.displayName)
+      return React.cloneElement(accordionheader, { accordionPanelId: id });
+
+    return React.cloneElement(accordionheader);
+  });
+  return (
+    <details className={className} id={id}>
+      {childrenWithProps}
```

Comment:
Create a wrapper and place the AccordionHeader and User defined Children of Panel separately in a wrapper. That wrapper can be controlled with isExpended state that you will maintain here. You can use it to hide and show the content inside the Detail's body. 

Then you will not need any ploy-fill in unsupported browsers. .